Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Processes/ApiGatewayEmulatorProcess.cs

```diff
@@ -0,0 +1,80 @@
+using Amazon.Lambda.TestTool.Commands.Settings;
+using Amazon.Lambda.TestTool.Extensions;
+using Amazon.Lambda.TestTool.Models;
+using Amazon.Lambda.TestTool.Services;
+
+namespace Amazon.Lambda.TestTool.Processes;
+
+/// <summary>
+/// A process that runs the API Gateway emulator.
+/// </summary>
+public class ApiGatewayEmulatorProcess
+{
+    /// <summary>
+    /// The service provider that will contain all the registered services.
+    /// </summary>
+    public required IServiceProvider Services { get; init; }
+
+    /// <summary>
+    /// The API Gateway emulator task that was started.
+    /// </summary>
+    public required Task RunningTask { get; init; }
+
+    /// <summary>
+    /// The endpoint of the API Gateway emulator.
+    /// </summary>
+    public required string ServiceUrl { get; init; }
+
+    /// <summary>
+    /// Creates the Web API and runs it in the background.
+    /// </summary>
+    public static ApiGatewayEmulatorProcess Startup(RunCommandSettings settings, CancellationToken cancellationToken = default)
+    {
+        var builder = WebApplication.CreateBuilder();
+
+        builder.Services.AddApiGatewayEmulatorServices();
+        
+        var serviceUrl = $"http://{settings.Host}:{settings.ApiGatewayEmulatorPort}";
+        builder.WebHost.UseUrls(serviceUrl);
+        builder.WebHost.SuppressStatusMessages(true);
+
+        builder.Services.AddHealthChecks();
+
+        var app = builder.Build();
+
+        app.UseHttpsRedirection();
+
+        app.MapHealthChecks("/__lambda_test_tool_apigateway_health__");
+
+        app.Map("/{**catchAll}", (HttpContext context, IApiGatewayRouteConfigService routeConfigService) =>
+        {
+            var routeConfig = routeConfigService.GetRouteConfig(context.Request.Method, context.Request.Path);
+            if (routeConfig == null)
+            {
+                context.Response.StatusCode = StatusCodes.Status403Forbidden;
+                context.Response.Headers.Append("x-amzn-errortype", "MissingAuthenticationTokenException");
+                return Results.Json(new { message = "Missing Authentication Token" });
+            }
+            
+            if (settings.ApiGatewayEmulatorMode.Equals(ApiGatewayEmulatorMode.HttpV2))
+            {
+                // TODO: Translate to APIGatewayHttpApiV2ProxyRequest
+            }
+            else
+            {
+                // TODO: Translate to APIGatewayProxyRequest
+            }
+
+            return Results.Ok();
+        });
+
+        var runTask = app.RunAsync(cancellationToken);
+        
+        return new ApiGatewayEmulatorProcess
```

Comment:
At this point do a `LogInformation` that the API Gateway emulator is available at the serviceUrl endpoint. Imagine this is what non-Aspire users will learn where to send requests to.

Reply:
This is a bit redundant since the dotnet process will print the following to the console:
```
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: **http://localhost:5051**
dbug: Microsoft.AspNetCore.Hosting.Diagnostics[13]
      Loaded hosting startup assembly Amazon.Lambda.TestTool
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: /Users/asmarp/CodeBase/aws-lambda-dotnet/Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool
dbug: Microsoft.Extensions.Hosting.Internal.Host[2]
      Hosting started
```

Reply:
Also, at this point in time, the process may not successfully start. So if I was to print the service URL and then the task failed, it would be kind of weird. I'd prefer if we rely on the default .NET logging for this.

Reply:
The ASP.NET Core log line below doesn't give users context that this is the API Gateway emulator versus Lambda runtime API.
```
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: **http://localhost:5051**
```

I believe in the published version of the tool we will default all of the Microsoft logging default level to `Error` so that our console window is focused on giving logging messages focused on our tool. When users see the console of our tool that should be thinking this is a log message for a lambda emulator tool not a web server.

So we need somewhere in the output that says basically the API Gateway Emulator endpoint is here. If this isn't that right spot that is fine but needs to be somewhere. 

Imagine a user installing who just does `dotnet install lambda-test-tool` and `dotnet lambda-test-tool`. There mind set should only be how to I test my Lambda function and what is informative to me as a Lambda developer debugging. Not how to a debug some webserver, that is our implementation detail.

Reply:
I added an informational message when the application starts using:
```
app.Lifetime.ApplicationStarted.Register(() =>
{
    app.Logger.LogInformation("The API Gateway Emulator is available at: {ServiceUrl}", serviceUrl);
});
```